File: app/src/main/java/com/cookplan/recipe/import_recipe/parser/ParserResultListener.java
package com.cookplan.recipe.import_recipe.parser;

import com.cookplan.models.Ingredient;
import com.cookplan.models.Recipe;

import java.util.List;
import java.util.Map;

/**
 * Created by DariaEfimova on 09.06.17.
 */

public interface ParserResultListener {

    void onSuccess(Recipe recipe, Map<String, List<Ingredient>> ingredientList);

    void onError(String error);
}

File: app/src/main/java/com/cookplan/recipe/import_recipe/parser/ParserResultListener.kt
package com.cookplan.recipe.import_recipe.parser

import com.cookplan.models.Ingredient
import com.cookplan.models.Recipe

/**
 * Created by DariaEfimova on 09.06.17.
 */

interface ParserResultListener {

    fun onSuccess(recipe: Recipe, ingredientList: MutableMap<String, List<Ingredient>>)

    fun onError(error: String)
}